index.d.ts

```diff
@@ -633,6 +634,22 @@ declare namespace plugins {
    */
   interface amqplib extends Instrumentation {}
 
+  /**
+   * This plugin automatically instruments the
+   * [aws-sdk](https://github.com/aws/aws-sdk-js) module.
+   */
+  interface aws_sdk extends Instrumentation {
+    /**
+     * Hooks to run before spans are finished.
+     */
+    hooks?: {
+      /**
+       * Hook to execute just before the aws span finishes.
+       */
+      request?: (span?: opentracing.Span, response?: anyObject) => any;
```

Comment:
Is the type definition for the Response available anywhere? If se we could copy it to provide the type.

Reply:
We do, i'm not particularly with how to do this in ts though, I just bulk copy/pasta this definition? there are some other aws specific types it relies on as well https://github.com/aws/aws-sdk-js/blob/5b51b4e3297f0f1f13d66eaff0c2e8ad5473de49/lib/response.d.ts#L3

Reply:
Yes, unfortunately this is the only way to do it for libraries that don't have a separate package for the types.

Reply:
As discussed, if this ends up being too complicated because of the way aws-sdk type definitions are exposed, let's leave `anyObject` for now.